File: online-judge-solutions/Codeforces/1284A.py
#https://codeforces.com/problemset/problem/1284/A
#Lang		:	Python 3.8
#Time		:	46 ms
#Memory         :	0 KB
n,m=map(int,input().split())
s= list(map(str,input().strip().split()))[:n]
t= list(map(str,input().strip().split()))[:m]
q=int(input())
for i in range(q):
    a=int(input())
    print(s[((a%n)+(n-1))%n]+t[((a%m)+(m-1))%m])

File: online-judge-solutions/Codeforces/1284A.cpp
//https://codeforces.com/problemset/problem/1284/A
//Lang		:	GNU C++14
//Time		:	31 ms
//Memory	:	0 KB
#include<iostream>
#include<vector>
using namespace std;
int main(){
	int n,m,a,q;
	cin>>n>>m;
	vector<string> s,t;
	string x;
	for(int i=0;i<n;i++){cin>>x;s.push_back(x);}
	for(int i=0;i<m;i++){cin>>x;t.push_back(x);}
	cin>>q;
	while(q--){
		cin>>a;
		cout<<s[((a%n)+(n-1))%n]+t[((a%m)+(m-1))%m]<<endl;
	}
	return 0;
}
